drivers/media/i2c/irs1125.c

```diff
@@ -614,8 +613,23 @@ static int irs1125_s_ctrl(struct v4l2_ctrl *ctrl)
 	}
 	case IRS1125_CID_MOD_PLL: {
 		struct irs1125_mod_pll *mod_new;
+		u8 *pval;
+		bool valid;
+		int s;
 
+		s = sizeof(struct irs1125_mod_pll) * IRS1125_NUM_MOD_PLLS;
+		valid = false;
 		mod_new = (struct irs1125_mod_pll *)ctrl->p_new.p;
+
+		for (pval = (u8 *)mod_new; pval < (u8 *)mod_new + s; pval++) {
+			if (*pval != 0) {
+				valid = true;
+				break;
+			}
+		}
+		if (!valid)
+			break;
```

Comment:
Do we not want to return an error if invalid?

Reply:
I now skip only writing this control in case of internal sync.